pkgs/development/tools/devpod/default.nix

```diff
@@ -0,0 +1,185 @@
+            stdenv-darwin = "sha256-z1dPaS5g6xlzKEJzPkpeShB+dTJHgQbzjQ9Rq3o+Xjo=";
+          }."${stdenv.name}";
+        };
+
+        buildPhase = ''
+          export HOME=$(mktemp -d)
+          yarn --offline run build
+
+          cp -r deps/devpod/dist $out
+        '';
+
+        distPhase = "true";
+        dontInstall = true;
+      };
+
+      rustTargetPlatformSpec = rust.toRustTargetSpec stdenv.hostPlatform;
+    in
+    rustPlatform.buildRustPackage {
+      inherit version src;
+      pname = "devpod-desktop";
+
+      sourceRoot = "${src.name}/desktop/src-tauri";
+
+      cargoLock = {
+        lockFile = "${src}/desktop/src-tauri/Cargo.lock";
+        outputHashes = {
+          "tauri-plugin-log-0.1.0" = "sha256-Ei0j7UNzsK45c8fEV8Yw3pyf4oSG5EYgLB4BRfafq6A=";
+        };
+      };
+
+      # Workaround:
+      #   The `tauri` dependency features on the `Cargo.toml` file does not match the allowlist defined under `tauri.conf.json`.
+      #   Please run `tauri dev` or `tauri build` or add the `updater` feature.
+      patches = [ ./add-tauri-updater-feature.patch ];
+
+      postPatch = ''
+        ln -s ${cli}/bin/devpod bin/devpod-cli-${rustTargetPlatformSpec}
+        cp -r ${frontend-build} frontend-build
+
+        substituteInPlace tauri.conf.json --replace '"distDir": "../dist",' '"distDir": "frontend-build",'
+      '' + lib.optionalString stdenv.isLinux ''
+        substituteInPlace $cargoDepsCopy/libappindicator-sys-*/src/lib.rs \
+          --replace "libayatana-appindicator3.so.1" "${libayatana-appindicator}/lib/libayatana-appindicator3.so.1"
+
+        # Since `cargo build` is used instead of `tauri build`, configs are merged manually.
+        jq --slurp '.[0] * .[1]' tauri.conf.json tauri-linux.conf.json >tauri.conf.json.merged
+        mv tauri.conf.json.merged tauri.conf.json
+      '';
+
+      nativeBuildInputs = [
+        copyDesktopItems
+        pkg-config
+      ] ++ lib.optionals stdenv.isLinux [
+        jq
+      ] ++ lib.optionals stdenv.isDarwin [
+        desktopToDarwinBundle
+      ];
+
+      buildInputs = [
+        libsoup
+        openssl
+      ] ++ lib.optionals stdenv.isLinux [
+        gtk3
+        libayatana-appindicator
+        webkitgtk
+      ] ++ lib.optionals stdenv.isDarwin [
+        darwin.apple_sdk.frameworks.Carbon
+        darwin.apple_sdk.frameworks.Cocoa
+        darwin.apple_sdk.frameworks.WebKit
+      ];
+
+      desktopItems = [
+        (makeDesktopItem {
+          name = "DevPod";
+          categories = [ "Development" ];
+          comment = "Spin up dev environments in any infra";
+          desktopName = "DevPod";
+          exec = "DevPod %U";
+          icon = "DevPod";
+          terminal = false;
+          type = "Application";
+          mimeTypes = [ "x-scheme-handler/devpod" ];
+        })
+      ];
+
+      postInstall = ''
+        ln -sf ${cli}/bin/devpod $out/bin/devpod-cli
+        mv $out/bin/devpod-desktop $out/bin/DevPod
```

Comment:
Is the CamelCase intended? The `devpop-{cli,desktop}` pattern strikes me as good tab-completion experience, while the CamelCase one would be poor for those with case insenitive tab completion enabled.

Reply:
Yes, it is intended, it is consistent with upstream naming